persistence/block.go

```diff
@@ -80,13 +83,21 @@ func (p *PostgresContext) prepareBlock(proposerAddr, quorumCert []byte) (*coreTy
 		txs[i] = txResult.GetTx()
 	}
 
+	// Retrieve the network id from the config
+	networkId := viper.GetString("network_id")
```

Comment:
viper should never be exposed outside of `runtime_configs`. You need to propagate it via `Config` and store it locally in `Create`